postoffice_django/publishing.py

```diff
@@ -85,21 +85,28 @@ class BulkPublisher(Publisher):
     def __init__(self, topic, payload, **attributes):
         super().__init__(topic=topic, payload=payload, bulk=True, **attributes)
 
-    def _create_message(self) -> list:
-        return [{
+    def _create_message(self) -> dict:
+        return {
             'topic': self.topic,
-            'payload': message,
+            'payload': self.payload,
             'attributes': self._stringify_attributes()
-        } for message in self.payload]
+        }
 
     def _create_publishing_error(self, error: str) -> None:
         PublishingError.objects.create(
             topic=self.topic,
-            payload=self.message,
+            payload=self._create_failed_message(),
             attributes=self.attributes,
             bulk=self.bulk,
             error=error,
         )
 
+    def _create_failed_message(self) -> list:
```

Comment:
This, to me, is quite risky. If I want to publish 1 million messages in bulk, and it fails, it will then take some time for PostOffice to get all those messages and do its thing. Can we save it in bulk, maybe in the future, and make Postoffice be able to receive it in bulk too?

Reply:
We're saving just 1 message from our side, and PostOffice will process it as a bulk message

Reply:
Oh, we are? I thought that because of the `for` we were saving more than one. Sorry.